ocs_ci/ocs/machine.py

```diff
@@ -138,6 +140,113 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
+                    'machine.openshift.io/cluster-api-cluster'
+                ] = cls_id
+                machineset_yaml['spec']['template']['metadata']['labels'][
+                    'machine.openshift.io/cluster-api-machine-role'
+                ] = role
+                machineset_yaml['spec']['template']['metadata']['labels'][
+                    'machine.openshift.io/cluster-api-machine-type'
+                ] = role
+                machineset_yaml['spec']['template']['metadata']['labels'][
+                    'machine.openshift.io/cluster-api-machineset'
+                ] = f"{cls_id}-{role}-{aws_zone}"
+                machineset_yaml['spec']['template']['spec'][
+                    'metadata'
+                ]['labels'][f"node-role.kubernetes.io/{role}"] = f"{label}"
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value']['ami']['id'] = ami_id
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'iamInstanceProfile'
+                ]['id'] = f"{cls_id}-worker-profile"
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'instanceType'
+                ] = instance_type
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'placement'
+                ]['availabilityZone'] = aws_zone
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'placement'
+                ]['region'] = region
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'securityGroups'
+                ][0]['filters'][0]['values'][0] = f"{cls_id}-worker-sg"
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'subnet'
+                ]['filters'][0]['values'][0] = f"{cls_id}-private-{aws_zone}"
+                machineset_yaml['spec']['template']['spec']['providerSpec']['value'][
+                    'tags'
+                ][0]['name'] = f"kubernetes.io/cluster/{cls_id}"
+                logging.info(f"Print yaml {machineset_yaml}")
+
+                # Create new custom machineset
+                ms_obj = OCS(**machineset_yaml)
+                ms_obj.create()
+    else:
+        raise UnsupportedPlatformError("Functionality not supported in UPI")
+
+
+def delete_custom_machineset(machine_set):
+    """
+    Function to delete custom machineset
+
+    Args:
+        machine_set (str): Name of the machine set to be deleted
+        WARN: Make sure it's not OCS worker node machines set, if so then
+              OCS worker nodes and machine set will be deleted.
+
+    Returns:
+        bool: True if commands executes successfully
+    """
+    ocp = OCP(namespace=constants.OPENSHIFT_MACHINE_API_NAMESPACE)
+    ocp.exec_oc_cmd(f'delete machineset {machine_set}')
+    return True
```

Comment:
return after validating that the machine set is deleted.

Reply:
Added validation in function, resolving comment.